src/components/app/create/CustomTagSelector/TagLabelValueSelector.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useState, useEffect } from 'react'
+import { OptionType, TagType } from '../../../app/types'
+import { PopupMenu } from '../../../common'
+import { ValidationRules } from '../validationRules'
+import { ReactComponent as ErrorCross } from '../../../../assets/icons/ic-close.svg'
+import { ReactComponent as Info } from '../../../../assets/icons/ic-info-outlined.svg'
+
+export default function TagLabelValueSelector({
+    selectedTagIndex,
+    tagData,
+    setTagData,
+    tagOptions,
+    isRequired,
+    type,
+}: {
+    selectedTagIndex: number
+    tagData: TagType
+    setTagData: (index: number, tagData: TagType) => void
+    tagOptions?: OptionType[]
+    isRequired?: boolean
+    type?: string
+}) {
+    const [selectedValue, setSelectedValue] = useState<string>('')
+    const [isPopupOpen, togglePopup] = useState<boolean>(false)
+
+    const validationRules = new ValidationRules()
+
+    useEffect(() => {
+        const _tagData = { ...tagData }
+        setSelectedValue(_tagData[type])
```

Comment:
Do we need to create const here as we'll just be accessing tagData type? Also, can't it be done where selectedTagIndex or tagData is being changed instead of doing it in useEffect?

Reply:
Removed the var. Yes we need to do it here to avoid unnecessary re rendering